src/main/java/com/epam/aidial/core/config/FileConfigStore.java

```diff
@@ -24,7 +23,7 @@ public final class FileConfigStore implements ConfigStore {
 
     private final String[] paths;
     private volatile Config config;
-    private final Map<String, String> deploymentKeys = new HashMap<>();
+    private final Map<String, ApiKeyData> keys = new HashMap<>();
```

Comment:
Please move it to a difference class. Config store just loads a config. Config should be immutable. Managing keys is a different responsibility and should be done outside.

Reply:
fixed